tests/phpunit/tests/media.php

```diff
@@ -5173,64 +5405,71 @@ public function test_wp_get_loading_optimization_attributes_low_res_image() {
 	}
 
 	/**
+	 * Tests that the `do_shortcode` context results in a lazy-loaded image by default.
+	 *
 	 * @ticket 58681
+	 * @ticket 58853
 	 *
-	 * @dataProvider data_wp_get_loading_optimization_attributes_in_shortcodes
+	 * @covers ::wp_get_loading_optimization_attributes
 	 */
-	public function test_wp_get_loading_optimization_attributes_in_shortcodes( $setup, $expected, $message ) {
+	public function test_wp_get_loading_optimization_attributes_in_shortcodes() {
 		$attr = $this->get_width_height_for_high_priority();
-		$setup();
 
-		// The first image processed in a shortcode should have fetchpriority set to high.
+		// Shortcodes processed outside of content blobs like 'the_content' always get `loading="lazy"`.
 		$this->assertSame(
-			$expected,
+			array(
+				'loading' => 'lazy',
+			),
 			wp_get_loading_optimization_attributes( 'img', $attr, 'do_shortcode' ),
-			$message
+			'Lazy-loading not applied to shortcodes outside the loop.'
 		);
 	}
 
-	public function data_wp_get_loading_optimization_attributes_in_shortcodes() {
-		return array(
-			'main_shortcode_image_should_have_fetchpriority_high'  => array(
-				'setup'    => function () {
-					global $wp_query;
+	/**
+	 * Tests that the `do_shortcode` context does not result in loading optimization changes when used within a content
+	 * blob.
+	 *
+	 * @ticket 58853
+	 * @dataProvider data_get_filters_with_do_shortcode_callback
+	 *
+	 * @covers ::wp_get_loading_optimization_attributes
```

Comment:
```suggestion
	 * @ticket 58853
	 *
	 * @covers ::wp_get_loading_optimization_attributes
	 *
	 * @dataProvider data_get_filters_with_do_shortcode_callback
```

Update annotation position per [56070](https://core.trac.wordpress.org/ticket/56070) proposal.